src/posix/platform/netif.cpp

```diff
@@ -1095,7 +1095,8 @@ static void processReceive(otMessage *aMessage, void *aContext)
     }
 }
 
-#if OPENTHREAD_CONFIG_NAT64_TRANSLATOR_ENABLE || OPENTHREAD_CONFIG_BORDER_ROUTING_DHCP6_PD_ENABLE
+#if OPENTHREAD_CONFIG_NAT64_TRANSLATOR_ENABLE || \
+    (OPENTHREAD_CONFIG_BORDER_ROUTING_ENABLE && OPENTHREAD_CONFIG_BORDER_ROUTING_DHCP6_PD_ENABLE)
```

Comment:
is it necessary to guard them by the macros? Could we remove them in a future PR?

Reply:
Yes, or the compilers will complain about unused methods.